backend/capellacollab/projects/capellamodels/modelsources/git/models.py

```diff
@@ -88,34 +73,16 @@ class DB_GitModel(Base):
     password = Column(String)
 
     @classmethod
-    def from_new_git_source(cls, model_id: int, source: NewGitSource):
+    def from_post_git_model(
+        cls, model_id: int, primary: bool, new_model: PostGitModel
+    ):
         return cls(
             name="",
-            path=source.path,
-            entrypoint=source.entrypoint,
-            revision=source.revision,
-            primary=True,
+            path=new_model.path,
+            entrypoint=new_model.entrypoint,
+            revision=new_model.revision,
+            primary=primary,
             model_id=model_id,
-            username=source.username,
-            password=source.password,
+            username=new_model.username,
+            password=new_model.password,
```

Comment:
This could definitely use a structure like this for more concision (That’s not a bug tho).
```python
return cls(
  name="",
  primary=primary,
  model_id=model_id,
  **new_model.dict(),
)
```